pac4j-core/src/main/java/org/pac4j/core/http/ajax/DefaultAjaxRequestResolver.java

```diff
@@ -13,21 +14,36 @@
  * @since 1.8.0
  */
 public class DefaultAjaxRequestResolver implements AjaxRequestResolver, HttpConstants, Pac4jConstants {
-
+    
+    private boolean addRedirectionUrlAsHeader = false;
+    
     @Override
     public boolean isAjax(final WebContext context) {
         final boolean xmlHttpRequest = AJAX_HEADER_VALUE.equalsIgnoreCase(context.getRequestHeader(AJAX_HEADER_NAME));
         final boolean hasDynamicAjaxParameter = Boolean.TRUE.toString().equalsIgnoreCase(context.getRequestHeader(IS_AJAX_REQUEST));
         final boolean hasDynamicAjaxHeader = Boolean.TRUE.toString().equalsIgnoreCase(context.getRequestParameter(IS_AJAX_REQUEST));
         return xmlHttpRequest || hasDynamicAjaxParameter || hasDynamicAjaxHeader;
     }
-
+    
     @Override
-    public HttpAction buildAjaxResponse(final RedirectionAction action, final WebContext context) {
+    public HttpAction buildAjaxResponse(WebContext context, RedirectionActionBuilder redirectionActionBuilder) {
```

Comment:
I find this code a bit complicated. I'd prefer to have:

```java
    public HttpAction buildAjaxResponse(final WebContext context, final RedirectionActionBuilder redirectionActionBuilder) {
        String url = null;
        if (addRedirectionUrlAsHeader) {
            final RedirectionAction action = redirectionActionBuilder.redirect(context);
            if (action instanceof FoundAction) {
                url = ((FoundAction) action).getLocation();
            }
        }

        if ( CommonHelper.isBlank(context.getRequestParameter(FACES_PARTIAL_AJAX_PARAMETER))) {
            if (CommonHelper.isNotBlank(url)) {
                context.setResponseHeader(HttpConstants.LOCATION_HEADER, url);
            }
            throw UnauthorizedAction.INSTANCE;
        }

        final StringBuilder buffer = new StringBuilder();
        buffer.append("<?xml version='1.0' encoding='UTF-8'?>");
        buffer.append("<partial-response>");
        buffer.append("<redirect url=\"" + url.replaceAll("&", "&amp;") + "\"></redirect>");
        buffer.append("</partial-response>");

        return new OkAction(buffer.toString());
    }
```

Reply:
I tried to keep the old method to avoid hard compatibility issues, but if you don't think it's worth it, that's all right! I did a commit with your version, just adding a `if` on the partial builder if url is null. Please, have a look. :)

Reply:
I got your idea, but as it's a major version, we can really change the source code.